libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -1,88 +1,78 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
-import { Store, select } from '@ngrx/store';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
 import {
   addToReadingList,
   clearSearch,
   getAllBooks,
   ReadingListBook,
-  searchBooks,
-  getBooksError
+  searchBooks
 } from '@tmo/books/data-access';
 import { FormBuilder } from '@angular/forms';
 import { Book } from '@tmo/shared/models';
-import { takeUntil } from 'rxjs/operators';
 import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, filter, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
   styleUrls: ['./book-search.component.scss']
 })
 export class BookSearchComponent implements OnInit, OnDestroy {
-  public books: ReadingListBook[];
-  public ngUnsubscribe$: Subject<boolean> = new Subject();
-  public errorMessage = '';
-  public readonly getBooks$ = this.store.pipe(select(getAllBooks));
-  public readonly getBooksError$ = this.store.pipe(select(getBooksError));
+  books: ReadingListBook[];
+  public displayBooks = false;
+  public unsubscriber: Subject<any> = new Subject<void> ();
 
   searchForm = this.fb.group({
     term: ''
   });
-
   constructor(
     private readonly store: Store,
     private readonly fb: FormBuilder
   ) {}
-
   get searchTerm(): string {
     return this.searchForm.value.term;
   }
-
   ngOnInit(): void {
-    this.getBooks$.pipe(takeUntil(this.ngUnsubscribe$)).subscribe(books => {
+    this.store.select(getAllBooks).subscribe(books => {
       this.books = books;
-      this.errorMessage = ''
     });
-    this.getBooksError$.pipe(takeUntil(this.ngUnsubscribe$)).subscribe((data) => {
-      if (data) {
-        this.errorMessage = 'Invalid Book Name'
-        this.books = []
-      }
-    });
-    this.searchForm.valueChanges.pipe(takeUntil(this.ngUnsubscribe$))
-      .subscribe(() => {
-        if (this.searchTerm === '') {
-          this.errorMessage = ''
-          this.books = []
-        }
-      });
+
+    this.instantSearchOnValueChange();  
   }
 
-  // formatDate(date: void | string) {
-  //   return date
-  //     ? new Intl.DateTimeFormat('en-US').format(new Date(date))
-  //     : undefined;
-  // }
+  instantSearchOnValueChange(){
+    this.searchForm.get('term').valueChanges.pipe(
+      takeUntil(this.unsubscriber),
+      debounceTime(500),
+      distinctUntilChanged(),
+      filter(data => data !== '')
```

Comment:
When your local is up please test this... I have a little doubt on the condition of this code. Also you might be able to avoid the filter altogether and put the condition inside distinctUntilChanged

Reply:
Tested the implementation on local. working fine.